src/startupCheck.ts

```diff
@@ -1,24 +1,27 @@
-import { window, commands, Uri } from 'vscode';
+import { window, commands, Uri, ExtensionContext, Extension } from 'vscode';
 
 import { VSCodeCommands } from './commands';
+import { ConfigurationConstants } from './constants';
 import { checkSupportedDotnetVersion } from './dotnet';
 import { Messages } from './ui/messages';
 
-export default async function checkAndInformAboutInstallation(): Promise<boolean> {
-  return await checkDotnetInstallation();
+export default async function checkAndInformAboutInstallation(context: ExtensionContext): Promise<boolean> {
+  return await checkDotnetInstallation(context);
 }
 
-async function checkDotnetInstallation(): Promise<boolean> {
+async function checkDotnetInstallation(context: ExtensionContext): Promise<boolean> {
   const answer = await checkSupportedDotnetVersion();
   if(answer !== undefined) {
     const selection = await window.showErrorMessage(
       answer,
       Messages.Dotnet.ChangeConfiguration,
       Messages.Dotnet.VisitDownload
     );
+    const extension = (context as unknown as { extension: Extension<void> }).extension;
```

Comment:
This looks like you're using a private API. Is there a way to do this using a public API?

Reply:
Good point.
This is actually documented public API, but for an unknown reason it's not exposed in VSCode.
https://code.visualstudio.com/api/references/vscode-api#ExtensionContext
It's not even written as "deprecated" and I did not find any alternative to access the extension's id (except maybe 'parsing' the URI, which isn't robust as in development mode the URI does not contain the ID)

Reply:
I can write `context.extension.id` without issue.

Reply:
What experiment did you run so that you don't have issue? Here is my VSCode
![image](https://user-images.githubusercontent.com/3601079/176206894-17be8963-2d3d-4cc2-a1d9-1e912378e9d8.png)

And after fixing the warning in the CI, we get the same error:
> [tsl] ERROR in /home/runner/work/ide-vscode/ide-vscode/src/startupCheck.ts(23,25)
      TS2339: Property 'extension' does not exist on type 'ExtensionContext'.

Do you have any other idea about why this happens, or should I revert your commit?